File: smallpict-java/src/main/java/com/smallpict/errors/ValidationException.java
package com.smallpict.errors;

import java.util.Map;

public class ValidationException extends SmallPictException {
    public ValidationException(String message) {
        this(message, null, null);
    }

    public ValidationException(String message, String requestId, Map<String, Object> details) {
        super(message, "VALIDATION_FAILED", 400, requestId, details, null);
    }
}

File: smallpict-php/src/Exceptions/ValidationException.php
<?php

declare(strict_types=1);

namespace SmallPict\Exceptions;

class ValidationException extends SmallPictException
{
    /**
     * @param string $message
     * @param string|null $requestId
     * @param array<string, mixed> $details
     * @param \Throwable|null $previous
     */
    public function __construct(
        string $message = 'Validation failed',
        ?string $requestId = null,
        array $details = [],
        ?\Throwable $previous = null
    ) {
        parent::__construct($message, 'VALIDATION_FAILED', 400, $requestId, $details, $previous);
    }
}
